sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala

```diff
@@ -2558,6 +2558,14 @@ abstract class SQLQuerySuiteBase extends QueryTest with SQLTestUtils with TestHi
       }
     }
   }
+
+  test("SPARK-32347: cte hint should be resolved in Hints batch rule") {
+    withTempView("t") {
+      sql("create temporary view t as select 1 as id")
+      sql("with cte as (select /*+ BROADCAST(id) */ id from t) select id from cte")
```

Comment:
Just in case, could you check that the hist is correctly applied?

Reply:
@maropu Sure, we can see in branch 2.4, the hint of cte could be resolved in `CTESubstitution`, we have a chance `executeSameContext` to apply all rules to CTE include hint resolve, but in branch 3.0, the `CTESubstitution` is reparated, and the `executeSameContext` is belong to `Analyzer`

Reply:
Thank you, I need to check the hist, seems something wrong with the patch.